go/logger/buffer.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package logger
+
+import (
+	"bufio"
+	"io"
+	"sync"
+	"time"
+)
+
+type autoFlushingBufferedWriter struct {
+	lock           sync.RWMutex
+	bufferedWriter *bufio.Writer
+	backupWriter   *bufio.Writer
+
+	frequency        time.Duration
+	timer            *time.Timer
+	currentlyWaiting bool
+	shutdown         chan struct{}
+}
+
+var _ io.Writer = &autoFlushingBufferedWriter{}
+
+func (writer *autoFlushingBufferedWriter) backgroundFlush() {
+	for {
+		select {
+		case <-writer.timer.C:
+			// Swap out active and backup writers
+			writer.lock.Lock()
+			writer.currentlyWaiting = false
+			previouslyActive := writer.bufferedWriter
+			writer.bufferedWriter = writer.backupWriter
```

Comment:
You can do `writer.bufferedWriter, writer.backupWriter = writer.backupWriter, writer.bufferedWriter`.